test/controllers/tasks_controller_test.rb

```diff
@@ -103,22 +109,65 @@
     # Note:  If there was a way to fail to save the changes to a task, that would be a great
     #        thing to test.
     it "can update an existing task" do
-      # Your code here
+      # Arrange
+      task_hash = {
+          task: {
+              name: "new name",
+              description: "new description"
+          }
+      }
+      task
+      expect{
+        patch task_path(task.id), params: task_hash
+      }.must_differ "Task.count", 0
+
+      updated_task = Task.find_by(id: task.id)
+      expect(updated_task.name).must_equal task_hash[:task][:name]
+      expect(updated_task.description).must_equal task_hash[:task][:description]
     end
     
     it "will redirect to the root page if given an invalid id" do
-      # Your code here
+      get task_path(-1)
+
+      # Assert
+      must_respond_with :redirect
     end
   end
   
   # Complete these tests for Wave 4
   describe "destroy" do
-    # Your tests go here
-    
+    it "successfully deletes an existing Task" do
+      existing_task = Task.last
+
+      expect {
+        delete task_path(existing_task.id)
+      }.must_change "Task.count", -1
```

Comment:
```suggestion
    it "successfully deletes an existing Task" do
      task.save

      expect {
        delete task_path(task.id)
      }.must_change "Task.count", -1
```
I noticed this test and another were erroring out for the same reason and I didn't understand why so I decided to look into it.
It turns out `let` syntax does not properly update the test database so that task from the `let` block isn't there. 
If you change the code to what I have here, the test passes because the task is actually added to the test database.

Reply:
I hope you didn't spend too much time on this. It's a really funky thing that we didn't ever explain!
It would really be better if the test we gave you had used a `before` block instead because then the test database would've been populated as you expected.